clippy_lints/src/bytecount.rs

```diff
@@ -86,7 +87,8 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for ByteCount {
                                            "Consider using the bytecount crate",
                                            format!("bytecount::count({}, {})",
                                                     snippet(cx, haystack.span, ".."),
-                                                    snippet(cx, needle.span, "..")));
+                                                    snippet(cx, needle.span, "..")),
+                                           Applicability::HasPlaceholders);
```

Comment:
It only "maybe" has placeholders. We should make the `snippet` function return a tuple of `Applicability` and snippet-`String`

Reply:
The best thing to do here would be to have it accept `&mut Applicability` IMO